### DISCUSSION TITLE
Is go to variable definition supported?

### DISCUSSION THREAD

**Discussion by dmi7ry** [JohnnyMorganz/luau-lsp#393] [Q&A]
Hi.
Is go to variable definition supported?

**Comment by JohnnyMorganz** [comment]
Hey, yes it should be supported.
I haven't given it a lot of attention though - let me know if you think it needs improving!

**Comment by dmi7ry** [comment]
It seems that it's broken (partially at least).
I'm trying in a new file
```
local test = "hello"
print(test)
```
I click F12 at `test` in the second line and it says `no definition found`.
And sometime it does'n work for tables and even functions (not sure what it depends).

**Reply by JohnnyMorganz** [reply]
Hm, just tried this now and it worked for me
If you have a larger example, I can take a look

**Reply by JohnnyMorganz** [reply]
I found that no language features actually seem to work in new unsaved files, and opened an issue: https://github.com/JohnnyMorganz/luau-lsp/issues/394

I don't think this is the problem you are facing though, because I think you are working with saved files

**Reply by dmi7ry** [reply]
Yep. You can see that file is saved, on the video I posted. BTW, have you tried the example I posted?

**Reply by JohnnyMorganz** [reply]
> Yep. You can see that file is saved, on the video I posted. BTW, have you tried the example I posted?

Yeah, I tried the example. I could not reproduce the issue (only could reproduce when it was a new untitled file - I have fixed this)

**Reply by dmi7ry** [reply]
Then we will try dig it and report if find any new details.

**Comment by dmi7ry** [comment]
For me:
1. Open two files (actually I have opened about 20 lua/luau files)
2. Restart VSCode
3. Try F12 on a variable. It works.
4. Switch to another tab. Try F12 on a variable. It doesn't work.
5. If make a change in the file and save it, it may start to work. And oposite, after do some changes, it may stop to work.

I'm not sure even it depeneds on content of the file. Alhough an example I used:
```
local test = "hello"
print(test)

local tbl = {
    hello = "hello"
}
print(tbl)

function foo()
end
foo()
test = "world"
```

Sometimes it works fine. Sometimes it may work partially (works for functions only). Sometimes doesn't work at all.
In the example if I change `foo` to `fo`, it may stop to work.

**Comment by dmi7ry** [comment]
Small video that shows the last example https://drive.google.com/file/d/1ORFo0bI4W-xhWBlJFTrQH5ukmQFpevnO/view?usp=sharing
